src/lib/token.test.ts

```diff
@@ -13,314 +14,523 @@ import {
+  zkAppBAddress = zkAppBKey.toPublicKey();
+  zkAppB = new ZkAppB(zkAppBAddress, tokenId);
 
-  tokenAccount2Key = Local.testAccounts[2].privateKey;
-  tokenAccount2 = tokenAccount2Key.toPublicKey();
+  zkAppCKey = Local.testAccounts[2].privateKey;
+  zkAppCAddress = zkAppCKey.toPublicKey();
+  zkAppC = new ZkAppC(zkAppCAddress, tokenId);
+}
 
-  let tx = await Mina.transaction(feePayer, () => {
-    AccountUpdate.fundNewAccount(feePayer);
-    zkapp.init();
-    zkapp.deploy({ zkappKey });
+async function setupLocal() {
+  setupAccounts();
+  let tx = await Mina.transaction(feePayerKey, () => {
+    let feePayerUpdate = AccountUpdate.createSigned(feePayerKey);
+    feePayerUpdate.balance.subInPlace(Mina.accountCreationFee().mul(1));
+    feePayerUpdate.send({
+      to: tokenZkappAddress,
+      amount: Mina.accountCreationFee(),
+    });
+    tokenZkapp.deploy();
+    tokenZkapp.init();
   });
+  await tx.prove();
+  tx.sign([tokenZkappKey]);
+  await tx.send();
+}
 
+async function setupLocalProofs() {
+  setupAccounts();
+  zkAppC = new ZkAppC(zkAppCAddress, tokenId);
+  let tx = await Mina.transaction({ feePayerKey }, () => {
+    let feePayerUpdate = AccountUpdate.createSigned(feePayerKey);
+    feePayerUpdate.balance.subInPlace(Mina.accountCreationFee().mul(3));
+    feePayerUpdate.send({
+      to: tokenZkappAddress,
+      amount: Mina.accountCreationFee(),
+    });
+    tokenZkapp.deploy();
+    tokenZkapp.init();
+    tokenZkapp.deployZkapp(zkAppBAddress, ZkAppB._verificationKey!);
+    tokenZkapp.deployZkapp(zkAppCAddress, ZkAppC._verificationKey!);
+  });
+  await tx.prove();
+  tx.sign([tokenZkappKey, zkAppBKey, zkAppCKey]);
   await tx.send();
 }
 
 describe('Token', () => {
-  beforeAll(async () => await isReady);
+  beforeAll(async () => {
+    await isReady;
+    await TokenContract.compile();
+    await ZkAppB.compile();
+    await ZkAppC.compile();
+  });
 
   afterAll(() => setTimeout(shutdown, 0));
 
-  describe('Create existing token', () => {
-    beforeEach(async () => {
-      await setupLocal();
-    });
+  describe('Signature Authorization', () => {
+    describe('Token Contract Creation/Deployment', () => {
+      beforeEach(async () => {
+        await setupLocal();
+      });
 
-    it('should have a valid custom token id', async () => {
-      const tokenId = zkapp.experimental.token.id;
-      const expectedTokenId = new Token({ tokenOwner: zkappAddress }).id;
-      expect(tokenId).toBeDefined();
-      expect(tokenId).toEqual(expectedTokenId);
-    });
+      test('correct token id can be derived with an existing token owner', () => {
+        expect(tokenId).toEqual(Token.getId(tokenZkappAddress));
+      });
 
-    it('should have a valid token symbol', async () => {
-      const symbol = Mina.getAccount(zkappAddress).tokenSymbol;
-      expect(tokenSymbol).toBeDefined();
-      expect(symbol).toEqual(tokenSymbol);
-    });
+      test('deployed token contract exists in the ledger', () => {
```

Comment:
can you organize the tests and comment how they map to the token tests doc? Similar to https://github.com/o1-labs/snarkyjs/blob/main/src/examples/zkapps/voting/test.ts#L54 . This will help us double-check that we covered everything in the doc -- and I think you mentioned you added a few more cases that weren't covered in the doc, can you add it there too?